stdlib/public/Glibc/CMakeLists.txt

```diff
@@ -3,44 +3,57 @@ set(sources
-endif()
-if (NOT EXISTS "${GLIBC_ARCH_INCLUDE_PATH}/sys")
-  message(FATAL_ERROR "Glibc headers were not found.")
-endif()
+      # Determine the location of glibc based on the target.
+      set(GLIBC_INCLUDE_PATH "/usr/include")
+      set(GLIBC_ARCH_INCLUDE_PATH "${GLIBC_INCLUDE_PATH}")
+      if(("${SDK}" STREQUAL "LINUX" OR "${SDK}" STREQUAL "FREEBSD") AND CMAKE_LIBRARY_ARCHITECTURE)
+        # FIXME: Some distributions install headers in
+        #        "/usr/include/x86_64-linux-gnu/sys/...". Here we use the host
+        #        machine's path, regardless of the SDK target we're building for.
+        #        This will break if cross-compiling from a distro that uses the
+        #        architecture as part of the path to a distro that does not.
+        set(GLIBC_ARCH_INCLUDE_PATH "${GLIBC_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+      endif()
 
-# Generate module.map
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-configure_file(module.map.in "${CMAKE_CURRENT_BINARY_DIR}/module.map" @ONLY)
-endif()
-if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
-configure_file(module.freebsd.map.in "${CMAKE_CURRENT_BINARY_DIR}/module.map" @ONLY)
-endif()
+      # Configure the module map based on the target. Each platform needs to
+      # reference different headers, based on what's available in their glibc.
+      set(modulemap_path "${CMAKE_CURRENT_BINARY_DIR}/${SWIFT_SDK_${SDK}_LIB_SUBDIR}/${arch}/module.map")
+      if("${SDK}" STREQUAL "FREEBSD")
+        configure_file(module.freebsd.map.in "${modulemap_path}" @ONLY)
+      else()
+        configure_file(module.map.in "${modulemap_path}" @ONLY)
+      endif()
 
-add_custom_command_target(unused_var
-    COMMAND
-        "${CMAKE_COMMAND}" "-E" "make_directory" "${output_dir}"
-    COMMAND
-        "${CMAKE_COMMAND}" "-E" "copy_if_different"
-        "${CMAKE_CURRENT_BINARY_DIR}/module.map"
-        "${output_dir}/module.map"
-    CUSTOM_TARGET_NAME "copy_glibc_module"
-    OUTPUT "${output_dir}/module.map" "${output_dir}"
-    DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/module.map"
-    COMMENT "Copying Glibc module to ${output_dir}")
+      set(VARIANT_SUFFIX "-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-${arch}")
+      add_custom_command_target(unused_var
+          COMMAND
+              "${CMAKE_COMMAND}" "-E" "make_directory" "${output_dir}"
+          COMMAND
+              "${CMAKE_COMMAND}" "-E" "copy_if_different"
+              "${modulemap_path}"
+              "${output_dir}/glibc.modulemap"
+          CUSTOM_TARGET_NAME "copy_glibc_module${VARIANT_SUFFIX}"
+          OUTPUT "${output_dir}/glibc.modulemap" "${output_dir}"
+          DEPENDS "${modulemap_path}"
+          COMMENT "Copying Glibc module to ${output_dir}")
 
-swift_install_in_component(stdlib
-    FILES "${output_dir}/module.map"
-    DESTINATION "lib/swift/glibc")
+      swift_install_in_component(stdlib
+          FILES "${output_dir}/glibc.modulemap"
+          DESTINATION "${output_dir}")
```

Comment:
Awesome, thanks @rintaro! I guess your removal of this directive in #1757 resolved the issue.